### PR TITLE
added contacts controller with CRUD operations

### PR DESCRIPTION
**Pull request by elliotfouts**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "if-env": "^1.0.4",
     "jquery": "^3.5.1",
     "moment": "^2.26.0",
+    "mongodb": "^3.5.8",
```
Comment (millernj) [inline package.json:39]: Why are we installing mongodb? All of our interfacing with mongoDB should be managed by mongoose.

File: controllers/contacts.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+const { Router } = require('express');
+const { Contact } = require('../models');
+const { dropUndefined } = require('../utils/dropUndefined');
+const { buildFilter } = require('../utils/queryHelper');
+const ObjectId = require('mongodb').ObjectID;
```
Comment (millernj) [inline controllers/contacts.js:5]: You can also reference ObjectId with `mongoose.ObjectId`.

File: controllers/contacts.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+const router = Router();
+
+// gets one contact by contact ID
+router.get('/:id', async (req, res) => {
+	const { id } = req.params;
+	let contactId = ObjectId(id);
+
+	try {
+		const contact = await Contact.findById(contactId);
```
Comment (millernj) [inline controllers/contacts.js:15]: Entering a string representation of a document's ObjectId into mongoose's `findById` method automatically converts the string into an `ObjectId`.

File: controllers/contacts.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	}
+});
+
+// gets all contacts associated with a user
+router.get('/userid/:id', async (req, res) => {
+	const { id } = req.params;
+	let contactId = ObjectId(id);
+
+	try {
+		const contact = await Contact.find({ userId: contactId });
```
Comment (millernj) [inline controllers/contacts.js:33]: The conversion to `ObjectId` for the inputed `userId` could be unnecessary here.

Comment (elliotfouts) [reply controllers/contacts.js:33]: I was having issues with that last night. Mongoose was returning null when I searched with a string ID and it seemed to only be working after the conversion to ObjectID. However, I'll go through it again in a second. Its definitely possible that something else was amiss

Comment (elliotfouts) [reply controllers/contacts.js:33]: nevermind im a kook. Fixing it right now

**Comment by millernj** [comment]
@elliotfouts Did removing the conversion to ObjectId work?

File: controllers/contacts.js (hunk 4 above)
Comment (millernj) [reply controllers/contacts.js]: Weird, any idea what could've been wrong last night?

Comment (elliotfouts) [reply controllers/contacts.js]: I'm not entirely sure but it had to be some unrelated issue that I caught later on. Good catch on that

**Comment by elliotfouts** [comment]
Yeah it works without the conversion

**Pull request merged**